insights/parsers/satellite_compute_resources.py

```diff
@@ -0,0 +1,50 @@
+"""
+SatelliteComputeResources - command ``psql -d foreman -c 'select name, type from compute_resources'``
+=====================================================================================================
+
+The SatelliteComputeResources parser reads the output of
+``psql -d foreman -c 'select name, type from compute_resources'``.
+
+Sample output of ``psql -d foreman -c 'select name, type from compute_resources'``::
+
+              name          |          type
+    ------------------------+-------------------------
+    test_compute_resource1 | Foreman::Model::Libvirt
+    test_compute_resource2 | Foreman::Model::RHV
+    (2 rows)
+
+Examples::
+
+    >>> type(resources)
+    <class 'insights.parsers.satellite_compute_resources.SatelliteComputeResources'>
+    >>> len(resources)
+    2
+    >>> resources['test_compute_resource1']
+    'Foreman::Model::Libvirt'
+"""
+
+from insights import parser, CommandParser
+from insights.specs import Specs
+from insights.parsers import SkipException
+
+
+@parser(Specs.satellite_compute_resources)
+class SatelliteComputeResources(CommandParser, dict):
+    """
+    Read the ``psql -d foreman -c 'select name, type from compute_resources'``
+    and save the data to a dict.
+    """
+
+    def parse_content(self, content):
+        columns = ['name', 'type']
+        if len(content) >= 4:
+            columns_in_data = [item.strip() for item in content[0].split('|')]
+            if columns_in_data == columns:
+                for row in content[2:]:
+                    if '|' in row:
+                        data = row.split('|')
+                        if len(data) != len(columns):
+                            raise SkipException("The output is in bad format")
+                        self[data[0].strip()] = data[1].strip()
+        if not self:
+            raise SkipException('No valid values')
```

Comment:
@huali027 however you simplify #2929, you might be able to take advantage of the same code to parse this data since they are both Postgresql output of specific items.  It might actually be more logical to have both this and #2929 in the same module since they are both satellite settings.

Reply:
@bfahr, sure, I'll combine them to the same module.